### ISSUE TITLE
Sporadic test errors since upgrading to Native 0.5 

### ISSUE THREAD

**Issue by armanbilge** [typelevel/cats#4647]
Most recently https://github.com/typelevel/cats/actions/runs/10556403923/job/29242594012

```
  [error] Error: Total 12066, Failed 0, Errors 12, Passed 12054
  [error] Error during tests:
  [error] 	cats.tests.MapSuite
  [error] 	cats.tests.ReducibleSuiteAdditional
  [error] 	cats.tests.BoundedEnumerableSuite
  [error] 	cats.tests.TraverseListSuiteUnderlying
  [error] 	cats.tests.NonEmptyAlternativeSuite
  [error] 	cats.tests.FunctorSuite
  [error] 	cats.tests.SemigroupKSuite
  [error] 	cats.tests.FunctionKLiftSuite
  [error] 	cats.tests.KleisliSuite
  [error] 	cats.tests.TupleSuite
  [error] 	cats.tests.IorSuite
  [error] 	cats.tests.WriterSuite
```

But this has been haunting us since we upgraded.

**Comment by danicheg** [comment]
Alright. What if we divide tests into two groups for Native build?

**Comment by armanbilge** [comment]
@danicheg Unfortunately the problem is likely a bug with MUnit or Scala Native itself. MUnit was hastily upgraded to multithreading.

**Comment by satorg** [comment]
I wonder – are there some other projects that suffer from the similar issue after upgrading to ScalaNative v5.x?
I only see it for Cats for now, but I may not be aware of all the projects around.

**Comment by WojciechMazur** [comment]
I'm not sure about how munit failed tests are reported, but when using JUnit in Scala Native project itself we got 2 categories: 
 - Failed tests - basically failed assertions 
 - Erronous tests - tests during execution of which a fatal error occoured, eg. segmenation fault 
 
 Based on the log above I think it touches the later category of fatal errors. It means we might need to investigate on the munit - scala native boundary

**Comment by armanbilge** [comment]
> It seems that when some tests fail, they do not fail because of any particular error – they just do not start:

Yes, I believe what is happening is that there are a number of test runners executing in parallel. Each runner is assigned some subset of the total test suites. It seems like if a test runner encounters a fatal error (eg segfault) in one of its suites, it dies, causing all other suites assigned to that runner to also be considered "errored".

**Comment by satorg** [comment]
@armanbilge , I wonder can those SEGFAULT errors be somehow related to tests that check for stack safety issues?

I mean, there are a plenty of tests in Cats that run quite memory-intense calculations just in order to make sure that there are no stack overflow errors. So I'm wondering – could such tests be actually the culprits?

**Comment by armanbilge** [comment]
If those tests were failing, and they were stackoverflowing, then that could manifest as a segfault. But if the tests are using constant stack space (as they should) then it should be an issue. It's worth checking out :)

**Comment by WojciechMazur** [comment]
That might be possible. Currently SN lacks proper handling for StackOverflowExceptions and OutOfMemoryError. The first one should be fixable by introducing canaries / signal handlers to recover from stack overflow. Similary we could try to handle OOM errors
